doc/upgrade-to-2.x.md

```diff
@@ -0,0 +1,474 @@
+| @opentelemetry/context-zone | 2.0.0 |
+| @opentelemetry/context-zone-peer-dep | 2.0.0 |
+| @opentelemetry/core | 2.0.0 |
+| @opentelemetry/exporter-jaeger | 2.0.0 |
+| @opentelemetry/exporter-zipkin | 2.0.0 |
+| @opentelemetry/propagator-b3 | 2.0.0 |
+| @opentelemetry/propagator-jaeger | 2.0.0 |
+| @opentelemetry/resources | 2.0.0 |
+| @opentelemetry/sdk-metrics | 2.0.0 |
+| @opentelemetry/sdk-trace-base | 2.0.0 |
+| @opentelemetry/sdk-trace-node | 2.0.0 |
+| @opentelemetry/sdk-trace-web | 2.0.0 |
+| @opentelemetry/shim-opentracing | 2.0.0 |
+| @opentelemetry/api-events | 0.200.0 |
+| @opentelemetry/api-logs | 0.200.0 |
+| @opentelemetry/exporter-logs-otlp-grpc | 0.200.0 |
+| @opentelemetry/exporter-logs-otlp-http | 0.200.0 |
+| @opentelemetry/exporter-logs-otlp-proto | 0.200.0 |
+| @opentelemetry/exporter-metrics-otlp-grpc | 0.200.0 |
+| @opentelemetry/exporter-metrics-otlp-http | 0.200.0 |
+| @opentelemetry/exporter-metrics-otlp-proto | 0.200.0 |
+| @opentelemetry/exporter-prometheus | 0.200.0 |
+| @opentelemetry/exporter-trace-otlp-grpc | 0.200.0 |
+| @opentelemetry/exporter-trace-otlp-http | 0.200.0 |
+| @opentelemetry/exporter-trace-otlp-proto | 0.200.0 |
+| @opentelemetry/instrumentation | 0.200.0 |
+| @opentelemetry/instrumentation-fetch | 0.200.0 |
+| @opentelemetry/instrumentation-grpc | 0.200.0 |
+| @opentelemetry/instrumentation-http | 0.200.0 |
+| @opentelemetry/instrumentation-xml-http-request | 0.200.0 |
+| @opentelemetry/opentelemetry-browser-detector | 0.200.0 |
+| @opentelemetry/otlp-exporter-base | 0.200.0 |
+| @opentelemetry/otlp-grpc-exporter-base | 0.200.0 |
+| @opentelemetry/otlp-transformer | 0.200.0 |
+| @opentelemetry/sampler-jaeger-remote | 0.200.0 |
+| @opentelemetry/sdk-events | 0.200.0 |
+| @opentelemetry/sdk-logs | 0.200.0 |
+| @opentelemetry/sdk-node | 0.200.0 |
+| @opentelemetry/shim-opencensus | 0.200.0 |
+| @opentelemetry/web-common | 0.200.0 |
+
+</details>
+
+## 💥 Node.js supported versions
+
+The **minimum supported Node.js has be raised to `^18.19.0 || >=20.6.0`**. This means that support Node.js 14 and 16 has been dropped.
+
+> [!NOTE]
+>
+> - The minimum supported Node.js versions for `@opentelemetry/api` (Node.js v8) and `@opentelemetry/semantic-conventions` (Node.js v14) are *not* changing as part of "JS SDK 2.x".
+> - The particular minimum *minor* versions of Node.js 18 and 20 were selected to include support for Node.js's `--import` flag and `module.register()` API. It is expected that this will provide a smoother experience for improved automatic ES module instrumentation.
+>
+> Related issues and PRs:
+> [#5395](https://github.com/open-telemetry/opentelemetry-js/issues/5395)
+
+## 💥 TypeScript supported versions
+
+The **minimum supported TypeScript version has been raised to 5.0.4**.
+
+As well, going forward all packages published from this repository will **drop support for old versions of `typescript` in minor releases**. We will only drop support for versions that are older than 2 years.
```

Comment:
Would it make sense to call out that this also `@opentelemetry/api` and `@opentelemetry/semantic-conventions` specifically? 🤔 Its implied by this but people may still be surprised by this.

Reply:
Yes, will do.

Reply:
commit 31529a640